src/EditorFeatures/Core/GoToDefinition/GoToDefinitionHelpers.cs

```diff
@@ -111,6 +156,63 @@ internal static class GoToDefinitionHelpers
                 project.Solution.Workspace, title, definitions).WaitAndGetResult(cancellationToken);
         }
 
+        public static bool TryGoToDefinitionOrParentDefinition(
+            ISymbol symbol,
+            Project project,
+            Document currentDocument,
+            int currentPosition,
+            IEnumerable<Lazy<IStreamingFindUsagesPresenter>> streamingPresenters,
+            CancellationToken cancellationToken,
+            bool thirdPartyNavigationAllowed = true,
+            bool throwOnHiddenDefinition = false)
+        {
+            var definitions = GetDefinitions(symbol, project, thirdPartyNavigationAllowed, cancellationToken);
+
+            if (definitions.Any(matchesCurrentPosition))
+            {
+                // If the current position is already on one of the definitions, we'll broaden the results
+                ImmutableArray<DefinitionItem> currentDefinition = default;
+                if (definitions.Length == 1)
+                {
+                    currentDefinition = definitions;
+                    definitions = ImmutableArray<DefinitionItem>.Empty;
+                }
+
+                var overridenDefinitions = GetOverridenDefinitions(symbol, project, thirdPartyNavigationAllowed, cancellationToken);
```

Comment:
overall logic here feels pretty confusing.  It seems like we simply want to say "get the main definitions.  if there are multiple or zero, do the normal logic.  If tehre is one, and we're located on it, then get the related definitions.  Then present handle the related definitions using the normal logic."